Sources/Datadog/Core/Persistence/Files/Directory.swift

```diff
@@ -55,11 +55,14 @@ internal struct Directory {
         // Instead of iterating over all files and removing them one by one, we create a temporary
         // empty directory and replace source directory content with (empty) temporary folder.
         // This makes the deletion atomic, and is more performant in benchmarks.
-        let temporaryDirectory = try Directory(withSubdirectoryPath: "com.datadoghq/\(UUID().uuidString)")
-        try retry(times: 3, delay: 0.001) {
-            _ = try FileManager.default.replaceItemAt(url, withItemAt: temporaryDirectory.url)
+        let tempDirectory = try Directory(withSubdirectoryPath: "com.datadoghq/\(UUID().uuidString)")
+        do {
```

Comment:
Doesn't it change the existing logic entirely? Errors thrown from here are catched in a layer above (in data migrator) and get send to our Internal Monitoring system, for observability. Now it seems we just mute them here with no further consideration and measure, no?

Reply:
true, but i think there is already a problem with the current code:
```
try replaceItemAt(...) // throws and stops execution in case of error...
try? removeItem(...) // cleanup. but this line never executes in case of error as it stops above already
```

i proposed something in the other comment to keep propating the error and doing the cleanup at the same time 👇